entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/metamodel/AnnotationViewMappingReader.java

```diff
@@ -84,6 +85,11 @@ public ViewMapping readViewMapping(Class<?> entityViewClass) {
             viewMapping.setDefaultBatchSize(batchFetch.size());
         }
 
+        With[] with = entityViewClass.getAnnotationsByType(With.class);
```

Comment:
Use something like
```
Set<CteProvider> cteProviders = new LinkedHashSet<>();
for (With w : AnnotationUtils.getAllAnnotations(With.class)) {
  cteProviders.add(w.value());
}
Set<Withs> withs = AnnotationUtils.getAllAnnotations(Withs.class);
for (Withs w : withs) {
  for (With with : w.value()) {
  cteProviders.add(with.value());
  }
}
viewMapping.setCteProviders(cteProviders);
```

Reply:
Do you mean `Set<Class<? extends CteProvider>>` or che actual instance?

Reply:
Of course, I meant `Set<Class<? extends CteProvider>>`, sorry for that.

Reply:
About naming conventions: should I use `CteProvider` or `CTEPRovider`? I see `CTE` is all uppercase in other classes.

Reply:
I'll go with `CTEProvider`, to be consistent with `CTEBuilder`.